controllers/nutanixmachine_controller.go

```diff
@@ -565,60 +566,83 @@ func (r *NutanixMachineReconciler) checkFailureDomainStatus(rctx *nctx.MachineCo
 	}
 
 	// Set the NutanixMachine.status.failureDomain
-	rctx.NutanixMachine.Status.FailureDomain = &fdObj.Name
+	rctx.NutanixMachine.Status.FailureDomain = &fd
 
 	return nil
 }
 
-func (r *NutanixMachineReconciler) getFailureDomainObj(rctx *nctx.MachineContext, fdName string) (*infrav1.NutanixFailureDomain, error) {
+func (r *NutanixMachineReconciler) getFailureDomainInfo(rctx *nctx.MachineContext, fdName string) (*infrav1.NutanixFailureDomainSpec, error) {
```

Comment:
Consider rename the func name "getFailureDomainInfo" to "getFailureDomainSpec", since it returns the FD spec.